src/components/editor-page/editor-pane/table-extractor.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { Editor } from 'codemirror'
+import { createNumberRangeArray } from '../../common/number-range/number-range'
+import { insertAtCursor } from './tool-bar/utils/toolbarButtonUtils'
+
+export const isTable = (text: string): boolean => {
+  // Tables must consist of multiple rows and columns
+  if (!text.includes('\n') || !text.includes('\t')) {
+    return false
+  }
+  // Code within code blocks should not be parsed as a table
+  if (text.startsWith('```')) {
+    return false
+  }
+
+  const lines = text.split(/\r?\n/)
+                    .filter(line => line.trim() !== '')
+
+  // Tab-indented text should not be matched as a table
+  if (lines.every(line => line.startsWith('\t'))) {
+    return false
+  }
+  // Every line should have the same amount of tabs (table columns)
+  const tabsPerLines = lines.map(line => line.match(/\t/g)?.length ?? 0)
+  return tabsPerLines.every(line => line === tabsPerLines[0])
+}
+
+export const extractTable = (pasteData: string, pasteEditor: Editor): void => {
```

Comment:
This method is not only extracting the table, it also insert it into the editor. The name should reflect this. BUT, if you follow clean code, then you should know, that an "and" in the function-name is an indicator for "too much stuff is happening here". So you shouldn't insert the content into the editor here. This also increases the testability.

Reply:
Changed